flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointStatsTracker.java

```diff
@@ -86,9 +88,10 @@ public class CheckpointStatsTracker {
 
     private final JobID jobID;
     private final MetricGroup metricGroup;
+    private int totalNumberOfSubTasks;
 
-    /** The latest restored checkpoint. */
-    @Nullable private RestoredCheckpointStats latestRestoredCheckpoint;
+    private Optional<JobInitializationMetricsBuilder> jobInitializationMetricsBuilder =
```

Comment:
Why this? I think optional isn't even intended to be used for fields.

Reply:
Matter of preferences. `Optional`'s compile time checks > not very usable `@Nullable`'s warnings in our code base.